java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/db/DBEntityFromTables.java

```diff
@@ -0,0 +1,139 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.java.lsp.server.db;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.CodeActionParams;
+import org.eclipse.lsp4j.MessageParams;
+import org.eclipse.lsp4j.MessageType;
+import org.netbeans.api.db.explorer.ConnectionManager;
+import org.netbeans.api.db.explorer.DatabaseConnection;
+import org.netbeans.api.db.explorer.DatabaseException;
+import org.netbeans.api.java.project.JavaProjectConstants;
+import org.netbeans.api.progress.aggregate.BasicAggregateProgressFactory;
+import org.netbeans.api.progress.aggregate.ProgressContributor;
+import org.netbeans.api.project.Project;
+import org.netbeans.api.project.ProjectUtils;
+import org.netbeans.api.project.SourceGroup;
+
+import org.netbeans.modules.j2ee.persistence.api.entity.generator.EntitiesFromDBGenerator;
+import org.netbeans.modules.java.lsp.server.LspServerState;
+import org.netbeans.modules.java.lsp.server.protocol.CodeActionsProvider;
+import org.netbeans.modules.java.lsp.server.protocol.NbCodeLanguageClient;
+import org.netbeans.modules.java.lsp.server.protocol.QuickPickItem;
+import org.netbeans.modules.java.lsp.server.protocol.ShowInputBoxParams;
+import org.netbeans.modules.java.lsp.server.protocol.ShowQuickPickParams;
+import org.netbeans.modules.parsing.api.ResultIterator;
+import org.openide.util.Lookup;
+import org.openide.util.NbBundle;
+import org.openide.util.lookup.ServiceProvider;
+
+/**
+ *
+ * @author Jan Horvath
+ */
+@NbBundle.Messages({
+    "MSG_NoProject=No Project Open",
+    "MSG_NoDbConn=No DB Connection",
+    "MSG_NoSourceRoot=No source root found",
+    "MSG_SelectTables=Select Database Tables",
+    "MSG_EnterPackageName=Enter package name"
+})
+@ServiceProvider(service = CodeActionsProvider.class)
+public class DBEntityFromTables extends CodeActionsProvider {
+
+    private static final String COMMAND_ENTITY_FROM_TABLES = "db.entity.from.tables"; //NOI18N
+
+    @Override
+    public CompletableFuture<Object> processCommand(NbCodeLanguageClient client, String command, List<Object> arguments) {
+        if (!COMMAND_ENTITY_FROM_TABLES.equals(command)) {
+            return null;
+        }
+        Lookup.getDefault().lookup(LspServerState.class).openedProjects().thenAccept((projects) -> {
+            if (projects.length > 0) {
+                createEntityClassesInProject(client, projects[0]);
```

Comment:
Question: if the command is invoked with an active editor focused: shouldn't we create the classes in the editor's project instead in the 1st opened one ? The order of projects in `openedProjects` may be rather random.
Of course provided that the LSP command arguments contain some URI etc to begin with.